### ISSUE TITLE
ECS loop inside ECS loop not working.

### ISSUE THREAD

**Issue by agustinlopini** [dudaster/ele-custom-skin#388]
Hi, I'm trying to use Posts Widget + ECS loop inside another Posts Widget + ECS loop but for some reason the inner ECS loop is taking over the parent ECS loop.

I have two post types: Artists and Projects that by doing a bidirectional relationship using ACF I can set which project goes to any specific artist. My goal is to have a page where I list all artists and the last 3 project attached to each artists.

I started by doing a standard WP page called "Test" and added a Posts Widget that queries all Artists post type. As a template I selected an ECS loop template called "Artists Loop".
Inside that "Artist Loop" I added the post title (to display each artist name), and then a Posts Widget that queries all Projects which uses an ECS loop templated called "Projects Loop" which has the featured image, the post title (to display the project name), etc.

When I view the page "Test", the first artist will show up fine with its ECS "Artist Loop" style and its projects with their ECS "Projects Loop" styles, but then the following artists inherits the styles from the "Projects Loop".
Also, any shared widget between the two Posts Widgets, instead of displaying both shared widgets, it removes the widget from the parent, and it replaces its content on the children one.

If I switch the "Projects Loop" to any Elementor built-in template, everything works as expected.

So, it looks like the inner ECS loop never finishes looping in its own widget container and continues looping onto the parent ECS loop overwriting some stuff from it.

Thank you!

**Comment by dudaster** [comment]
I know currently the bested loops aren’t working. Try using the classic skin in the nested loops while I’ll find a solution to it.

**Comment by agustinlopini** [comment]
Yes, I'm still trying to debug it, I think the reason it breaks is due to how the `get_template()` function works in skin-custom.php file.
My guess is that since the `$wp_query` initally stores the query coming from the parent page (instead of the parent ECS loop), the second (and following items) are using incorrect data.

Ideally the first thing that we should do is store the WP_Query object coming from the actual Posts Widget query into `$old_query` and not the global `$wp_query` coming from the page. Then you can continue creating the new WP_Query to get the ECS loop page and return the `$old_query` at the end.

Another take is to never alter or store global `$wp_query` and just find a way to use the `$new_query` by itself but maybe this is not possible.

Thank you!

**Comment by trinermedia** [comment]
I am facing the same problem right now. Not a real solution, more of a workaround: How about implementing #69? Then instead of a nested post widget you could use the portfolio widget inside the custom loop skin.
